AdvLoggerPkg/Library/AdvancedLoggerLib/Pei/GoogleTest/AdvancedLoggerPeiLibGoogleTest.inf

```diff
@@ -0,0 +1,42 @@
+## @file AdvancedLoggerPeiLibGoogleTest.inf
+#
+#  Google Test for PEI instance of the Advanced Logger library.
+#
+#  Copyright (c) Microsoft Corporation.
+#  SPDX-License-Identifier: BSD-2-Clause-Patent
+#
+##
+
+[Defines]
+  INF_VERSION                    = 1.26
+  BASE_NAME                      = AdvancedLoggerPeiLibGoogleTest
+  FILE_GUID                      = 95196693-DA35-4775-887C-1FF9C65D5054
+  MODULE_TYPE                    = HOST_APPLICATION
+  VERSION_STRING                 = 1.0
+
+#
+# The following information is for reference only and not required by the build tools.
+#
+#  VALID_ARCHITECTURES           = IA32 X64 AARCH64
+#
+
+[Sources]
+  AdvancedLoggerPeiLibGoogleTest.cpp
+
+[Packages]
+  MdePkg/MdePkg.dec
+  UnitTestFrameworkPkg/UnitTestFrameworkPkg.dec
+  AdvLoggerPkg/AdvLoggerPkg.dec
+
+[LibraryClasses]
+  GoogleTestLib
+  AdvancedLoggerLib
+
+[Ppis]
+  gAdvancedLoggerPpiGuid                                    ## CONSUMES
+
+[Pcd]
+  gAdvLoggerPkgTokenSpaceGuid.PcdAdvancedLoggerHdwPortDebugPrintErrorLevel       ## SOMETIMES_CONSUMES

```

Comment:
@apop5 When is this PCD sometimes consumed? I copied it over from the Lib inf.

Reply:
![image](https://github.com/user-attachments/assets/8b7b7a7c-db6e-4d7c-9084-d411e9536b92)


Consider the case where there are multiple firmware images that were compiled separately. 

i.e. one project's DSC was compiled and then another project's DSC was compiled, and that between the two, they might result is a mismatch of the versions.

If the reported version of advanced logger in the system is newer than ADVANCED_LOGGER_HW_LVL_VER, it will not be used. Otherwise it will be used.